test/e2e/scheduling/priorities.go

```diff
@@ -465,16 +465,28 @@ func createBalancedPodForNodes(f *framework.Framework, cs clientset.Interface, n
 
 		needCreateResource[v1.ResourceMemory] = *resource.NewQuantity(int64((ratio-memFraction)*float64(memAllocatableVal)), resource.BinarySI)
 
-		err := testutils.StartPods(cs, 1, ns, string(uuid.NewUUID()),
-			*initPausePod(f, pausePodConfig{
-				Name:   "",
-				Labels: balancePodLabel,
-				Resources: &v1.ResourceRequirements{
-					Limits:   needCreateResource,
-					Requests: needCreateResource,
+		podConfig := &pausePodConfig{
+			Name:   "",
+			Labels: balancePodLabel,
+			Resources: &v1.ResourceRequirements{
+				Limits:   needCreateResource,
+				Requests: needCreateResource,
+			},
+			Affinity: &v1.Affinity{
+				NodeAffinity: &v1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{MatchExpressions: []v1.NodeSelectorRequirement{
+								{Key: "kubernetes.io/hostname", Operator: v1.NodeSelectorOpIn, Values: []string{node.Labels["kubernetes.io/hostname"]}},
```

Comment:
Please update this with `MatchFields` as well.

Reply:
done